src/mirrors_config.vala

```diff
@@ -116,16 +116,20 @@ namespace Pamac {
 						unowned Variant variant;
 						if (line.contains ("Method")) {
 							if (new_conf.lookup_extended ("Method", null, out variant)) {
-								data.append ("Method=%s\n".printf (variant.get_string ()));
+								if (variant.get_string () == "rank") {
+									data.append ("# Method = \n".printf (line));
+								} else {
+									data.append ("Method = %s\n".printf (variant.get_string ()));
+								}
 							} else {
 								data.append (line + "\n");
 							}
 						} else if (line.contains ("OnlyCountry")) {
 							if (new_conf.lookup_extended ("OnlyCountry", null, out variant)) {
 								if (variant.get_string () == "ALL") {
-									data.append ("#%s\n".printf (line));
+									data.append ("# OnlyCountry = \n".printf (line));
```

Comment:
This ends in:

```
mirrors_config.vala:120.23-120.51: error: Invalid type for argument 1
									data.append ("# Method = \n".printf (line));
									             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mirrors_config.vala:130.23-130.56: error: Too many arguments for specified format
									data.append ("# OnlyCountry = \n".printf (line));
									             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mirrors_config.vala:130.23-130.56: error: Invalid type for argument 1
```

The **%s** would be the current **line**. When you have an issue with `"OnlyCountry=%s\n"` and the fix would be `"OnlyCountry = %s\n"`. So you have to adopt similar from the second case to the fist case and actually test your code first.